p2p/src/sessions/mod.rs

```diff
@@ -372,3 +423,59 @@ pub struct GetConsolidatedPeersResult {
     /// The address shown here can be used to connect to this peers in the future.
     pub outbound: Vec<SocketAddr>,
 }
+
+/// A convenient wrapper around a collection of network ranges.
+/// This enables efficient tracking of the inbound connections we have so as to prevent sybil
+/// machines from monopolizing our inbound peers table.
+#[derive(Default)]
+pub struct NetworkRangesCollection {
+    inner: BTreeSet<Vec<u8>>,
```

Comment:
Unless we want the ip ranges to be sorted, using a `HashSet` is more efficient than using a `BTreeSet`:

```suggestion
    inner: HashSet<Vec<u8>>,
```